### PR TITLE
Split eth-types Error

### PR DESCRIPTION
**Pull request by silathdiir**
Close https://github.com/appliedzkp/zkevm-circuits/issues/294

Move `bus-mapping` related errors out of `eth-types`.

### CODE REVIEW

File: eth-types/src/error.rs [hunk 1]
```diff
@@ -14,10 +12,6 @@ pub enum Error {
     MemAddressParsing,
     /// Error while parsing a `StackAddress`.
     StackAddressParsing,
-    /// Error while trying to convert to an incorrect `OpcodeId`.
-    InvalidOpConversion,
```
Comment (HAOYUatHZ) [inline eth-types/src/error.rs:18]: `InvalidOpConversion` is a parsing error. should be in `eth-types`

File: eth-types/src/error.rs [hunk 2]
```diff
@@ -27,26 +21,6 @@ pub enum Error {
...
-    /// Error while generating a trace.
-    TracingError,
-    /// JSON-RPC related error.
-    JSONRpcError(ProviderError),
-    /// OpcodeId is not a call type.
-    OpcodeIdNotCallType,
-    /// Account not found in the StateDB
-    AccountNotFound(Address),
-    /// Storage key not found in the StateDB
-    StorageKeyNotFound(Address, Word),
```
Comment (HAOYUatHZ) [inline eth-types/src/error.rs:39]: where's  `AccountNotFound`? though we're not using it but we should not remove it. We should also put it in bus-mapping error

**Review by HAOYUatHZ — CHANGES_REQUESTED** [review]
after 
+ adding `InvalidOpConversion` to eth-types error and 
+ adding `StorageKeyNotFound` to bus-mapping error

will be great

**Review by HAOYUatHZ — APPROVED** [review]

**Comment by scroll-dev** [comment]
moved to https://github.com/appliedzkp/zkevm-circuits/pull/304

**Pull request closed without merging**